Review of mbed-os-atecc608a

GuyWi opened the pull request:
This PR is part of [IOTDOCS-1066](https://jira.arm.com/browse/IOTDOCS-1066).

Patater committed:
Enable SE support in Mbed Crypto

GuyWi committed:
Technical author review

GuyWi committed:
Technical author review

GuyWi commented:
@Patater, can you please review this PR? Thank you!

GuyWi commented on README.md:
```diff
@@ -1,2 +1,40 @@
 # Mbed OS Examples for the ATECC608A
 
```
@Patater, can we add an introductory sentence here?

Patater replied:
Something like the following?

"The examples contained in this repository demonstrate how to use the ATECC608A secure element with Mbed OS. The `atecc608a` example demonstrates use of the ATECC608A using Mbed Crypto. Additional examples for how to use generate a certificate using pre-provisioned ATECC608A keys, how to use ATECC608A with Mbed TLS, are yet to come."

GuyWi replied:
Yes, perfect! Thanks :)

k-stachowiak committed:
Rely on the public PSA API to reset security state

Commit:
Change the format of key slot printing

Commit:
Update the example to the new driver API.
The key slot numbers used in PSA currently map directly to the hardware slots.
Because of this, as slot 0 is reserved in PSA, slot 1 is used for private key
instead.
Change direct low-level driver calls to calls to the PSA API.
Remove the public key exporting during private key generation test, since this
functionality is not present anymore.
The storage is purged after the first key generation test for two reasons:
first - so that the same key slot can be reused and second - so that key
registration can be tested as a second way to have a key in a PSA slot.
Update the pointer to the new driver library in mbed-os-atecc608a.lib

Patater committed:
Merge pull request #10 from ARMmbed/psa-key-registration

Update the example to the new driver API

GuyWi committed:
Added intro provided by Jaeden.

GuyWi committed:
Added intro provided by Jaeden.

GuyWi commented on README.md:
````diff
@@ -1,2 +1,42 @@
+```sh
+git clone git@github.com:ARMmbed/mbed-os-example-atecc608a.git
+cd mbed-os-example-atecc608a/
+source ~/venvs/mbed/bin/activate
````
Do you need this line if you don't set up a virtual env?

Patater replied:
No, we don't need this line if we don't set up virtual environments. I don't think other examples mention virtual environments, so probably all mentioned of virtual environments should go away.

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi commented on README.md:
```diff
@@ -1,2 +1,42 @@
+
+## Hardware interface
+
+A couple of evaluation and development kits available for ATECC508A.
```
ATECC508A or ATECC608A?

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi commented on README.md:
```diff
@@ -1,2 +1,42 @@
+
+## Hardware interface
+
+A couple of evaluation and development kits are available for the ATECC508A secure element.
```
ATECC508A or ATECC608A?

Patater replied:
The chips are compatible; ATECC608A is a superset of ATECC508A. We don't use ATECC608A-only features. We can say ATECC608A here for consistency.

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi commented:
@Patater, I've updated the document. Could you please re-review?
Thanks! 
(However, I am still unable to compile the example.)

Commit:
Remove unnecessary script needed before mbed-os update
update-crypto.sh was used to setup a temporary and unreleased
version of Mbed Crypto and Mbed TLS libraries.

Commit:
Update the setup.md document
Sources have now been moved deeper in the directory structure
mbed new is not needed to compile the example

Patater commented on README.md:
````diff
@@ -1,2 +1,59 @@
+```sh
+git clone git@github.com:ARMmbed/mbed-os-example-atecc608a.git
+cd mbed-os-example-atecc608a/atecc608a
+mbed new .
````
I don't think the "mbed new ." step is required (and it might break things)

Patater commented on README.md:
````diff
@@ -1,2 +1,59 @@
+Your PC may take a few minutes to compile your code. At the end, you see the following result:
+
+```
+Image: ./BUILD/K64F/GCC_ARM/mbed-os-example-atecc608a.bin
````
When using the `--flash --sterm` options, you should see a bit more than this.

I see this.

```
Image: ./BUILD/K64F/GCC_ARM/atecc608a.bin
--- Terminal on /dev/ttyACM0 - 9600,8,N,1 ---
Serial Number:
01 23 BA CF BA D3 29 CA EE 

Config zone: 01 23 BA CF 00 00 50 00 BA D3 29 CA EE C0 3D 00 
C0 00 55 00 83 20 83 20 83 20 83 20 83 20 83 20 
83 20 83 20 00 00 00 00 00 00 00 00 00 00 00 00 
00 00 AF 8F FF FF FF FF 00 00 00 00 FF FF FF FF 
00 00 00 00 FF FF FF FF FF FF FF FF FF FF FF FF 
FF FF FF FF 00 00 00 00 FF FF 00 00 00 00 00 00 
13 00 13 00 13 00 13 00 13 00 13 00 13 00 13 00 
1C 00 10 00 10 00 10 00 10 00 10 00 10 00 1C 00 

--- Device locks information ---
  - Config locked: 1
  - Data locked: 1
  - Slot 0 locked: 0
  - Slot 1 locked: 0
  - Slot 2 locked: 0
  - Slot 3 locked: 0
  - Slot 4 locked: 0
  - Slot 5 locked: 0
  - Slot 6 locked: 0
  - Slot 7 locked: 0
  - Slot 8 locked: 0
  - Slot 9 locked: 0
  - Slot 10 locked: 0
  - Slot 11 locked: 0
  - Slot 12 locked: 0
  - Slot 13 locked: 0
  - Slot 14 locked: 0
  - Slot 15 locked: 0
--------------------------------

Private key slot in use: 1, public: 9
Running tests...
test_hash_sha256 succesful!
test_sign_verify succesful!
test_export_import succesful!
test_psa_import_verify succesful!
test_write_read_slot succesful!


Available commands:
 - info - print configuration information;
 - test - run all tests on the device;
 - exit - exit the interactive loop;
 - generate_private[=%d] - generate a private key in a given slot (0-15),
                           default slot - 0.
 - generate_public=%d_%d - generate a public key in a given slot
                           (0-15, second argument) using a private key
                           from a given slot (0-15, first argument);
 - private_slot=%d - designate a slot to be used as a private key in tests;
 - public_slot=%d - designate a slot to be used as a public key in tests;
 - write_lock_config - write a hardcoded configuration to the device,
                       lock it;
 - lock_data - lock the data zone;
```

Patater committed:
Merge pull request #13 from AndrzejKurek/script-docs-cleanup

Script and docs cleanup

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

Patater commented on README.md:
````diff
@@ -1,2 +1,115 @@
+mbed deploy
+```
+
+## Compiling the example
````
With `--flash --sterm`, this section is more accurately called "Compiling and running the example"

Patater commented on README.md:
```diff
@@ -1,2 +1,115 @@
+## Hardware interface
+
+A couple of evaluation and development kits are available for the ATECC608A secure element.
+To interface with an Mbed platform, you have to make inter-integrated circuit (I2C) and power supply connections. Note that ATECC508A requires a 5V supply.
```
"inter-integrated circult" is not what anybody ever writes. Please leave as I2C and don't expand it.

Patater commented on README.md:
```diff
@@ -1,2 +1,115 @@
+## Hardware interface
+
+A couple of evaluation and development kits are available for the ATECC608A secure element.
+To interface with an Mbed platform, you have to make inter-integrated circuit (I2C) and power supply connections. Note that ATECC508A requires a 5V supply.
```
Should we have a space between the 5 and the V? "5 V" or "5V"?

GuyWi replied:
Let's leave it 5V since this is consistent with the style guide (https://confluence.arm.com/display/TCOM/Units+of+measurement).

Patater replied:
OK

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

Patater commented:
This needs rebasing to resolve conflicts.

GuyWi committed:
Update README.md

GuyWi committed:
Update README.md

GuyWi committed:
Merge branch 'technical-author-review-of-readme' of https://github.com/ArmMbedCloud/mbed-os-example-atecc608a into technical-author-review-of-readme

GuyWi commented:
@Patater, I have rebased.

Patater commented:
> @Patater, I have rebased.

Oops, you merged. I will rebase for you.

GuyWi commented:
> > @Patater, I have rebased.
> 
> Oops, you merged. I will rebase for you.

Thank you :)

Patater commented:
I couldn't push to your branch. I made my own at https://github.com/ARMmbed/mbed-os-example-atecc608a/pull/14

GuyWi commented:
Thanks, @Patater! 
I'll close this PR, then.
